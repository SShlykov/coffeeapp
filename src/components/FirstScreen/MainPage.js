import React from "react";
import {BeansLogo} from "../SmallComp";

const FirstScreen = () => {
    return(
            <div className="row">
                <div className="col-lg-10 offset-lg-1">
                    <h1 className="title-big">Everything You Love About Coffee</h1>
                    <BeansLogo/>
                    <div className="preview__subtitle">
                        We makes every day full of energy and taste
                    </div>
                    <div className="preview__subtitle">Want to try our beans?</div>
                    <a href="#prev" className="preview__btn">
                        More
                    </a>
                </div>
            </div>
    )
};
export default FirstScreen;