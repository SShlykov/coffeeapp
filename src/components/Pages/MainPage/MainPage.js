import React, {Component} from "react";
import Item, {BeansLogo} from "../../SmallComp";

const MainPage = ({data}) => {
        const items = data.map(item=>{
            const {id, type="best", name, weight, price, imgSrc} = item;
            return <Item key={id} id={id} type={type} name={name} weight={weight} price={price} imgSrc={imgSrc}/>
        });
        return(
            <>
                <section className="about">
                    <div className="container">
                        <div className="row">
                            <div className="col-lg-6 offset-lg-3">
                                <div className="title">About Us</div>
                                <BeansLogo dark/>
                                <div className="about__text">
                                    Extremity sweetness difficult behaviour he of. On disposal of
                                    as landlord horrible. Afraid at highly months do things on at.
                                    Situation recommend objection do intention so questions. As
                                    greatly removed calling pleased improve an. Last ask him cold
                                    feel met spot shy want. Children me laughing we prospect
                                    answered followed. At it went is song that held help face.
                                    <br />
                                    <br />
                                    Now residence dashwoods she excellent you. Shade being under
                                    his bed her, Much read on as draw. Blessing for ignorant
                                    exercise any yourself unpacked. Pleasant horrible but confined
                                    day end marriage. Eagerness furniture set preserved far
                                    recommend. Did even but nor are most gave hope. Secure active
                                    living depend son repair day ladies now.
                                </div>
                            </div>
                        </div>
                    </div>
                </section>
                <section className="best">
                    <div className="container">
                        <div className="title">Our best</div>
                        <div className="row">
                            <div className="col-lg-10 offset-lg-1">
                                <div className="best__wrapper">
                                    {items}
                                </div>
                            </div>
                        </div>
                    </div>
                </section>
            </>
        )
};
export default MainPage;