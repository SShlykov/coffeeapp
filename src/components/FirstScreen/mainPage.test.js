import React from 'react';
import MainPage from "./MainPage";
import {shallow} from 'enzyme';

describe('Testing <MainPage/>', () => {
    const page = shallow(<MainPage id="1" type="best"/>);
    describe('Testing snap & state', () => {
        it('MainPage have rendered correctly', () => {
            expect(page).toMatchSnapshot();
        });
    });
});