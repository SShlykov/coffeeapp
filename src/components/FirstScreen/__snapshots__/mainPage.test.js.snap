// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <MainPage/> Testing snap & state MainPage have rendered correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FirstScreen
    id="1"
    type="best"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="col-lg-10 offset-lg-1"
      >
        <h1
          className="title-big"
        >
          Everything You Love About Coffee
        </h1>
        <BeansLogo />
        <div
          className="preview__subtitle"
        >
          We makes every day full of energy and taste
        </div>
        <div
          className="preview__subtitle"
        >
          Want to try our beans?
        </div>
        <Link
          className="preview__btn"
          replace={false}
          to="/ourcoffee"
        >
          More
        </Link>
      </div>,
      "className": "row",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="title-big"
          >
            Everything You Love About Coffee
          </h1>,
          <BeansLogo />,
          <div
            className="preview__subtitle"
          >
            We makes every day full of energy and taste
          </div>,
          <div
            className="preview__subtitle"
          >
            Want to try our beans?
          </div>,
          <Link
            className="preview__btn"
            replace={false}
            to="/ourcoffee"
          >
            More
          </Link>,
        ],
        "className": "col-lg-10 offset-lg-1",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Everything You Love About Coffee",
            "className": "title-big",
          },
          "ref": null,
          "rendered": "Everything You Love About Coffee",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "We makes every day full of energy and taste",
            "className": "preview__subtitle",
          },
          "ref": null,
          "rendered": "We makes every day full of energy and taste",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Want to try our beans?",
            "className": "preview__subtitle",
          },
          "ref": null,
          "rendered": "Want to try our beans?",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "More",
            "className": "preview__btn",
            "replace": false,
            "to": "/ourcoffee",
          },
          "ref": null,
          "rendered": "More",
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="col-lg-10 offset-lg-1"
        >
          <h1
            className="title-big"
          >
            Everything You Love About Coffee
          </h1>
          <BeansLogo />
          <div
            className="preview__subtitle"
          >
            We makes every day full of energy and taste
          </div>
          <div
            className="preview__subtitle"
          >
            Want to try our beans?
          </div>
          <Link
            className="preview__btn"
            replace={false}
            to="/ourcoffee"
          >
            More
          </Link>
        </div>,
        "className": "row",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="title-big"
            >
              Everything You Love About Coffee
            </h1>,
            <BeansLogo />,
            <div
              className="preview__subtitle"
            >
              We makes every day full of energy and taste
            </div>,
            <div
              className="preview__subtitle"
            >
              Want to try our beans?
            </div>,
            <Link
              className="preview__btn"
              replace={false}
              to="/ourcoffee"
            >
              More
            </Link>,
          ],
          "className": "col-lg-10 offset-lg-1",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Everything You Love About Coffee",
              "className": "title-big",
            },
            "ref": null,
            "rendered": "Everything You Love About Coffee",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "We makes every day full of energy and taste",
              "className": "preview__subtitle",
            },
            "ref": null,
            "rendered": "We makes every day full of energy and taste",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Want to try our beans?",
              "className": "preview__subtitle",
            },
            "ref": null,
            "rendered": "Want to try our beans?",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "More",
              "className": "preview__btn",
              "replace": false,
              "to": "/ourcoffee",
            },
            "ref": null,
            "rendered": "More",
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
